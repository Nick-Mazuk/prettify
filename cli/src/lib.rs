use prettify::print;
use prettify_markdown::format_markdown;
use std::time::Duration;

#[derive(PartialEq, Debug, Clone, Copy)]
pub enum Language {
    Markdown,
}

pub fn format_by_language(contents: &str, language: Language) -> String {
    print(match language {
        Language::Markdown => format_markdown(contents),
    })
}

pub fn get_language_from_filename(filename: &str) -> Option<Language> {
    if filename.ends_with(".md") {
        Some(Language::Markdown)
    } else {
        None
    }
}

pub fn get_elapsed_string(elapsed: Duration) -> String {
    if elapsed.as_millis() < 1 {
        format!(
            "{} microsecond{}",
            elapsed.as_micros(),
            if elapsed.as_micros() == 1 { "" } else { "s" }
        )
    } else if elapsed.as_secs() < 1 {
        format!(
            "{} millisecond{}",
            elapsed.as_millis(),
            if elapsed.as_millis() == 1 { "" } else { "s" }
        )
    } else {
        format!(
            "{} second{}",
            elapsed.as_secs(),
            if elapsed.as_secs() == 1 { "" } else { "s" }
        )
    }
}
