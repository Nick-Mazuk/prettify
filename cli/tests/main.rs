use cli::{format_by_language, get_language_from_filename};

pub fn format_file(file_name: &str, contents: &str) -> String {
    let language = get_language_from_filename(file_name);
    match language {
        Some(matched_language) => format_by_language(contents, matched_language),
        _ => contents.to_string(),
    }
}

#[test]
fn test_formatting() {
    insta::glob!("files/**/*.*", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let formatted = format_file(path.file_name().unwrap().to_str().unwrap(), &contents);
        insta::assert_snapshot!(formatted);
    });
}
