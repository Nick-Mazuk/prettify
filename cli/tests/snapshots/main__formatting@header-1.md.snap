---
source: cli/tests/main.rs
expression: formatted
input_file: cli/tests/files/markdown/header-1.md
---
# Hello world

